add endpoint for checking subdomain availability

groovecoder opened the pull request:
To test:

1. http://127.0.0.1:8000/accounts/profile/subdomain?subdomain=<subdomain-to-check>
   * Note: if your browser session isn't signed in, it should return an error
2. Also verify that you can still register a new subdomain

maxxcrawford reviewed (commented):
I tested the bad word filter with "butt" (http://127.0.0.1:8000/accounts/profile/subdomain?subdomain=butt) and it returned `true`. Should it return false, or is it baked into another query?

groovecoder commented:
Oh good catch! I didn't include the badwords check in there. Will fix and update.

groovecoder commented:
Okay I moved the "already taken" AND the "has bad word" check into the method that checks for availability. Should be fixed now. (Also added a couple tests)
